### PR TITLE
Add policy for CIS-1.1 - 4.04 Ensure oslogin is enabled for a Project

### PR DESCRIPTION
**Pull request by palani-ram-google-partner**
Add policy for CIS-1.1 - 4.04 Ensure oslogin is enabled for a Project

### CODE REVIEW

**Comment by palani-ram-google-partner** [comment]
@morgante Please review this PR.

File: policies/templates/gcp_compute_enable_oslogin_project_v1.yaml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+#
+apiVersion: templates.gatekeeper.sh/v1alpha1
+kind: ConstraintTemplate
+metadata:
+  name: gcp-compute-enable-oslogin-project-v1
+spec:
+  crd:
+    spec:
+      names:
+        kind: GCPComputeEnableOSLoginProjectConstraintV1
```
Comment (morgante) [inline policies/templates/gcp_compute_enable_oslogin_project_v1.yaml:23]: I think `GCPComputeRequireOSLoginConstraintV1` would be a better name.

File: samples/compute_enable_oslogin_project.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+apiVersion: constraints.gatekeeper.sh/v1alpha1
+kind: GCPComputeEnableOSLoginProjectConstraintV1
+metadata:
+  name: compute-enable-oslogin-project
+  annotations:
+    bundles.validator.forsetisecurity.org/cis-v1.1: 4.04
+    description: Checks if "oslogin" is enabled for a Project
```
Comment (morgante) [inline samples/compute_enable_oslogin_project.yaml:21]: Is this an accurate description? From the Rego, it looks like this actually checks that OS login is enabled on each VM in the proejct.

File: validator/compute_enable_oslogin_project.rego [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	asset := input.asset
+	asset.asset_type == "compute.googleapis.com/Instance"
+	instance := asset.resource.data
+	meta := lib.get_default(instance, "metadata", {"items": []})
+	key = "enable-oslogin"
+
+	# check if key is available and values are as expected
+	not metadata_enable_oslogin(meta)
+
+	message := sprintf("On this resource %v check the required key '%v' is in violation.", [asset.name, key])
```
Comment (morgante) [inline validator/compute_enable_oslogin_project.rego:36]: This message should be clearer about what actually went wrong (OS login not enabled for this VM).

File: validator/compute_enable_oslogin_project_test.rego [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+import data.test.fixtures.compute_enable_oslogin.constraints as fixture_constraints
+
+template_name := "GCPComputeEnableOSLoginProjectConstraintV1"
+
+#### Testing for GCE instances
+
+#1. No instances at all
+test_enable_oslogin_project_compute_no_instances {
+	expected_resource_names := {"//dns.googleapis.com/projects/186783260185/managedZones/correct"}
```
Comment (morgante) [inline validator/compute_enable_oslogin_project_test.rego:35]: This value isn't used anywhere.

File: validator/compute_enable_oslogin_project_test.rego [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+import data.validator.gcp.lib as lib
+import data.validator.test_utils as test_utils
+
+# Importing the test data
+import data.test.fixtures.compute_enable_oslogin.assets.compute.instance_no_violation as fixture_compute_instance_no_violation
+import data.test.fixtures.compute_enable_oslogin.assets.compute.instance_violation as fixture_compute_instance_violation
+import data.test.fixtures.compute_enable_oslogin.assets.compute.no_instances as fixture_compute_no_instance
+import data.test.fixtures.compute_enable_oslogin.assets.compute.no_metadata as fixture_compute_instance_no_metadata
+
+import data.test.fixtures.compute_enable_oslogin.constraints as fixture_constraints
```
Comment (morgante) [inline validator/compute_enable_oslogin_project_test.rego:27]:
```suggestion
import data.test.fixtures.compute_enable_oslogin.constraints as fixture_constraint
```

File: validator/compute_enable_oslogin_project_test.rego [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+#1. No instances at all
+test_enable_oslogin_project_compute_no_instances {
+	expected_resource_names := {"//dns.googleapis.com/projects/186783260185/managedZones/correct"}
+	test_utils.check_test_violations_count(fixture_compute_no_instance, [fixture_constraints], template_name, 0)
+}
+
+#2. One instance with correct key
+test_enable_oslogin_project_compute_instance_no_violations {
+	expected_resource_names := {"//compute.googleapis.com/projects/my-test-project/zones/us-central1-f/instances/test-jumphost"}
```
Comment (morgante) [inline validator/compute_enable_oslogin_project_test.rego:41]: This value isn't used.

File: validator/compute_enable_oslogin_project_test.rego [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+	test_utils.check_test_violations_count(fixture_compute_instance_no_violation, [fixture_constraints], template_name, 1)
+}
+
+#3. One instance without correct key
+test_enable_oslogin_project_compute_instance_violations {
+	expected_resource_names := {"//compute.googleapis.com/projects/my-test-project/zones/us-central1-f/instances/test-jumphost"}
+	test_utils.check_test_violations_count(fixture_compute_instance_violation, [fixture_constraints], template_name, 1)
```
Comment (morgante) [inline validator/compute_enable_oslogin_project_test.rego:48]: Should check the right resource was returned.

**Pull request merged**